### PR TITLE
SQSCANGHA-55 Add curl redirect and fix splatting of URL with special chars

### PR DESCRIPTION
**Pull request by antonioaversa**
This PR makes a few improvements on top of https://ngithub.com/SonarSource/sonarqube-scan-action/pull/151:
- `curl` should auto-redirect on 3XX
- `curl` should fail right away on non-2XX (except for redirect)
- splatting of `curl` and `wget` parameters should be done safely, to avoid the risk of command injection via custom SonarScanner URL

I couldn't actually find myself a scenario where the splatting would lead to commands injections (which would be done by the user on the user-owned environment anyway), since parameters are passed in the yaml and are always escaped by single quote (see [this example](https://github.com/SonarSource/sonarqube-scan-action/actions/runs/12068750215/job/33654545394?pr=154#step:5:57)).

Anyway, I have added QA tests to document the behavior.

### CODE REVIEW

File: .github/qa-sq-behind-ngix/nginx.conf
Comment (antonioaversa) [inline .github/qa-sq-behind-ngix/nginx.conf:1]: Changes in this file are unrelated. As I was testing nginx, I have removed unnecessary config.

**Review by henryju — APPROVED** [review]
LGTM!
